### PR TITLE
[mqtt.homeassistant] implement iif and is_defined jinja function and filters

### PR DESCRIPTION
**Pull request by ccutrer**
`iif` is pretty straight forward besides having optional parameters. `is_defined` is a bit weird, because it's _supposed_ to throw an exception, which is then silently ignored and used to _not_ update a value. This actually works out because mqtt.generic checks if a transformation returns null, and if it does just ignores the message.

There are lots more HASS extensions to implement, but it'll be a slow process to check which ones are already implemented in Jinjava, which ones don't even make sense in openHAB, which ones are even possible to implement, etc. But these two seem like the most desperately needed, and serve as a base to easily add more, so this is where I started.

Refs #15382, #17374, #16872.

For #16872, I need to do more investigation to see what's up. Jinjava will return an empty string for the template in that ticket, not `null`, so will set the item to `NULL`. (`null` will get ignored). The Home Assistant docs kind of imply that you should be using `is_defined` to get that sort of behavior, but I need to test if referencing a non-existent field in Python/Home Assistant will actually return `nil` as well.

### CODE REVIEW

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/HomeAssistantChannelTransformation.java [hunk 1]
```diff
@@ -89,8 +97,17 @@ public Optional<String> apply(String template, Map<String, @Nullable Object> bin
         try {
             transformationResult = jinjava.render(template, bindings);
         } catch (FatalTemplateErrorsException e) {
-            logger.warn("Applying template {} for component {} failed: {}", template,
-                    component.getHaID().toShortTopic(), e.getMessage());
+            var error = e.getErrors().iterator();
+            Exception exception = null;
```
Comment (lsiepel) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/HomeAssistantChannelTransformation.java:101]: Sure the first iteration always yields `UndefinedException` ?
Not needed to iterate all errors and 'search' for this specific type of exception ?

Comment (ccutrer) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/HomeAssistantChannelTransformation.java:101]: Yes. I don't remember the specific route off the top of my head, but it's a direct `throw`/`catch` between the undefined filter and here, the wrapping exception in Jinjava, which just makes sure it's an array. There's no opportunity for anything else to be added to that array for UndefinedException. Presumably there would be multiple elements for something like syntax errors.

**Review by lsiepel — APPROVED** [review]
LGTM
